backend/modules/publishing/service.py

```
@@ -1 +1,603 @@
+        temp_pdf_url = pdf_metadata.get("temp_pdf_url")
+        
+        # Check if PDF already exists in database
+        existing_file = self.repository.get_simulation_file(simulation.id, filename)
+        existing_case_study_url = getattr(simulation, 'case_study_url', None)
+        
+        if existing_case_study_url and _is_s3_url(existing_case_study_url):
+            logger.info(f"[PDF_STORAGE] Simulation already has case_study_url: {existing_case_study_url}")
+            return
+        
+        if existing_file and existing_file.file_path and _is_s3_url(existing_file.file_path):
+            logger.info(f"[PDF_STORAGE] PDF already exists in database with S3 URL")
```

Ruff (0.14.7)

f-string without any placeholders

Remove extraneous `f` prefix

(F541)